### ISSUE TITLE
Check for the presence of `fuse` as a **pre**condition.

### ISSUE THREAD

**Issue by ralienpp** [Nuitka/Nuitka#1509]
* Nuitka 0.7.6
* Installed via pip

I'm exploring the possibility of using Nuitka with a large project that takes about 40 minutes to compile. I used the `--onefile` parameter to see how it works out, and 40 minutes later: `FATAL: Error, the fuse library (libfuse .so.x from fuse2, *not* fuse3) must be installed for onefile creation to work on Linux.`

I suggest a small usability improvement - call `if not locateDLL("fuse"): ...` at the beginning, rather than at the end. It won't make a difference for small projects, but it will save time when there's a large project at hand and one is just familiarizing themselves with Nuitka and is not yet aware of all the prerequisites.

**Comment by ralienpp** [comment]
Update: after another cycle, I stopped at the next thing: `FATAL: Error, error exit from AppImage because fuse kernel module was not loaded`. The same rationale applies here, it would have been better Nuitka checked the prerequisites in advance and quit earlier.

**Comment by kayhayen** [comment]
I agree it would be better. It seems, though, that the AppImage mode may go away in the future. You can not use it, by the undocumented option `--onefile-tempdir` and then fuse is not needed. Predicting what AppImage will use relatively hard, e.g. the fuse library is OS dependent.

What's missing to make me switch is the ability to cache onefile content to folders, such that it will always be fast to run and needs no unpacking the second time around. But from what people tell me, even with temp folder usage, it seems to often be faster.

And inside Docker or something, it doesn't become easy to run fuse at all.

**Comment by kayhayen** [comment]
This unfortunately will have to wait until the discussed onefile improvements for the general code case happened.

**Comment by Kit-p** [comment]
> I agree it would be better. It seems, though, that the AppImage mode may go away in the future. You can not use it, by the undocumented option `--onefile-tempdir` and then fuse is not needed. Predicting what AppImage will use relatively hard, e.g. the fuse library is OS dependent.
> 
> What's missing to make me switch is the ability to cache onefile content to folders, such that it will always be fast to run and needs no unpacking the second time around. But from what people tell me, even with temp folder usage, it seems to often be faster.
> 
> And inside Docker or something, it doesn't become easy to run fuse at all.

@kayhayen Thank you for mentioning about the `--onefile-tempdir` option, but I used it together with `--include-package-data` and received the fatal error below that did not appear when using `--onefile`.

`FATAL: Error, package data files are only included in standalone or onefile mode.`

I am trying to use Nuitka in a CI pipeline and failed because Docker does not support loading the fuse kernel module. I would appreciate your help.
